### ISSUE TITLE
Allergen alert set in the previous version of the app (production) disappear in the beta

### ISSUE THREAD

**Issue by teolemon** [openfoodfacts/openfoodfacts-androidapp#4120]
An allergen alert set in the previous version of the app (production) disappears in the beta.

**Steps to reproduce:**
- I installed Open Food Facts as a normal user
- I added a gluten alert
- I tested that it worked
- I joined the beta (which is `develop`)
- I saw the gluten alert had disappeared
- I added a new one
- It worked again

**Potential causes**
```
2021-09-12 17:58:39.197 21409-21409/? D/Sentry: io.sentry.auto-init read: false
2021-09-12 17:58:39.197 21409-21409/? I/Sentry: Retrieving auto-init from AndroidManifest.xml
2021-09-12 17:58:39.211 21409-21409/? I/greenDAO: Migrating schema from version 18 to 22
2021-09-12 17:58:39.211 21409-21409/? E/MIGRATE VERSION: 19
2021-09-12 17:58:39.213 21409-21409/? E/MIGRATE VERSION: 20
2021-09-12 17:58:39.215 21409-21409/? E/MIGRATE VERSION: 21
2021-09-12 17:58:39.216 21409-21409/? E/MIGRATE VERSION: 22
2021-09-12 17:58:39.288 21409-21438/? I/Sentry: sentry-debug-meta.properties file was not found.
```
- [ ] [The file where migrations are done](https://github.com/openfoodfacts/openfoodfacts-androidapp/blob/develop/app/src/main/java/openfoodfacts/github/scrachx/openfood/utils/OFFDatabaseHelper.kt)
- [ ] https://github.com/openfoodfacts/openfoodfacts-androidapp/commit/0e06205848ed028061e346c9c1d23acca9a3c6ad#diff-5659840340e38dfaad346e16d4456a24a326f02a343e51331c7cf2d618b4da9e
- [ ] https://github.com/openfoodfacts/openfoodfacts-androidapp/commit/c558f1b29de9a8fcfaebb26b4ff62d41de505abd#diff-5659840340e38dfaad346e16d4456a24a326f02a343e51331c7cf2d618b4da9e

### Date of the production release (last known working)
Last updated Jan 14 8:16 PM
So roughly release 3.6.3 @ https://github.com/openfoodfacts/openfoodfacts-androidapp/commit/b691a6e9cae94dadb8659f51fae01d63c0e0f153

**Comment by teolemon** [comment]
There were 2 recent modifications to the DB schema @kartikaysharma01 @VaiTon Could any of those be causing this bug ?
- [ ] https://github.com/openfoodfacts/openfoodfacts-androidapp/commit/0e06205848ed028061e346c9c1d23acca9a3c6ad#diff-5659840340e38dfaad346e16d4456a24a326f02a343e51331c7cf2d618b4da9e
- [ ] https://github.com/openfoodfacts/openfoodfacts-androidapp/commit/c558f1b29de9a8fcfaebb26b4ff62d41de505abd#diff-5659840340e38dfaad346e16d4456a24a326f02a343e51331c7cf2d618b4da9e

**Comment by VaiTon** [comment]
@teolemon maybe you cleaned the data when running from the IDE. Could it be the case?

**Comment by teolemon** [comment]
Nope, I tested in real conditions:

- [ ] I installed Open Food Facts as a normal user
- [ ] I added a gluten alert
- [ ] I tested that it worked
- [ ] I joined the beta (which is `develop`)
- [ ] I saw the gluten alert had disappeared
- [ ] I added a new one
- [ ] It worked again

**Comment by teolemon** [comment]
@VaiTon nope, I think we need to:

- [ ] Print what's in the db before upgrade
- [ ] Print what's in the db after upgrade
- [ ] Print any exceptions
- [ ] If nothing convincing in the points above, move to see if it's a problem further down the line

**Comment by naivekook** [comment]
@teolemon production is master branch and beta is develop branch right?

**Comment by teolemon** [comment]
I don't think we merged into master. But I can tell you what date we released.

**Comment by teolemon** [comment]
Last updated Jan 14 8:16 PM

**Comment by teolemon** [comment]
So roughly release 3.6.3 @ https://github.com/openfoodfacts/openfoodfacts-androidapp/commit/b691a6e9cae94dadb8659f51fae01d63c0e0f153

**Comment by teolemon** [comment]
https://stackoverflow.com/questions/17529766/view-contents-of-database-file-in-android-studio

**Comment by naivekook** [comment]
So latest production release was on Jan 14 and somewhere from Jan 14 till now happens and database is broken for allergen alerts, quite a huge period to check

**Comment by naivekook** [comment]
the only one thing that changed for Allergen.kt and AllergenName.kt is the package renaming, don't know how it influent on greendao entities

**Comment by teolemon** [comment]
@naivekook we could just look at all 4 migrations, since they all seem to be erroring. I strongly doubt we messed up all 4.
Could it be related to something Kotlin-related ? 
I'm looking at this unrelated issue, for instance: https://stackoverflow.com/questions/57809370/database-migration-fail-after-adding-indents-with-kotlin-multiline-string-and-ro

**Comment by VaiTon** [comment]
@teolemon this should had been closed by #4325. Did we test? Can we close?

**Comment by teolemon** [comment]
I did test, and the fix by @g123k works 🤟

**Comment by VaiTon** [comment]
Fixed by #4325
